spec/system/consumer/shopping/shopping_spec.rb

```diff
@@ -330,24 +333,52 @@
       end
 
       context "filtering search results" do
-        it "returns no results and clears searches by clicking the clear-link" do
+        before do
           visit shop_path
           sleep(2)
+        end
+        it "returns no results and clears searches by clicking the clear-link" do
           fill_in "search", with: "74576345634XXXXXX"
           expect(page).to have_content "Sorry, no results found"
           expect(page).not_to have_content product2.name
           click_on "Clear search" # clears search by clicking text
           expect(page).to have_content("Add", count: 4)
         end
         it "returns results and clears searches by clicking the clear-button" do
-          visit shop_path
-          sleep(2)
           fill_in "search", with: "Meer" # For product named "Meercats"
           expect(page).to have_content product2.name
           expect(page).not_to have_content product.name
           find("a.clear").click # clears search by clicking the X button
           expect(page).to have_content("Add", count: 4)
         end
+        it "returns results by searching by keyword" do
+          # model: meta_keywords
+          fill_in "search", with: "Wild" # For product named "Meercats"
+          expect(page).to have_content product2.name
+        end
+        it 'returns results by searching by variants display_name' do
+          # model: variant display_name
+          fill_in "search", with: "Ferrets" # For variants named "Ferrets"
+          within('div.pad-top') do
+            expect(page).to have_content product2.variants[1].display_name # Ferrets
+            expect(page).not_to have_content product.variants[2].display_name # not Badgers
+          end
+        end
+        it 'returns results by searching by variants display_name' do
+          # model: variant display_as
+          fill_in "search", with: "displayedunder" # "Badgers"
+          within('div.pad-top') do
+            expect(page).not_to have_content product2.variants[1].display_name # Ferrets
+            expect(page).to have_content product.variants[2].display_name # not Badgers
+          end
+        end
+        it 'returns results by searching by variants display_name' do
+          # model: Enterprise name
+          fill_in "search", with: "Enterp" # Enterprise 1 sells nothing
+          within('p.no-results') do
+            expect(page).to have_content "Sorry, no results found for Enterp"
+          end
+        end
```

Comment:
Can we combine these `it` blocks? It's good practice to keep them separate as you did but it's very slow in reality and consumes unnecessary energy on every test run. CI can run 100 times a day.